File: 2024-dgu-spring-camp/01/3079.py
# 약수의 합
# https://study.helloalgo.co.kr/study/1019/room/588/6411

n = int(input())

answer = 0
# O(제곱근연산 + 루트N)
for i in range(1, int(n**0.5) + 1):
    if n % i == 0:
        answer += i
        if n // i != i:
            answer += n // i

print(answer)

File: 2024-dgu-spring-camp/01/3079.cpp
// 약수의 합
// https://study.helloalgo.co.kr/study/1019/room/588/6411

#include <iostream>
#include <cmath>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);
	
	long long n;
	long long answer = 0;
	
	cin >> n;
	
	for (long long i=1; i*i<=n; i++) {
		if (n % i == 0) {
			answer += i;
			if (n / i != i) {
				answer += (long long)(n / i);
			}
		}
	}
	
	cout << answer << endl;
	
	return 0;
}
